src/fhirtypes/ElementDefinition.ts

```diff
@@ -1537,7 +1537,7 @@ export class ElementDefinition {
       );
     } else if (type === 'Coding') {
       this.assignFHIRValue(code.toString(), code.toFHIRCoding(), exactly, 'Coding');
-    } else if (type === 'Quantity') {
+    } else if (type === 'Quantity' || type === 'Age') {
       // Since code only maps to part of Quantity, we want to ensure that if there are other (non-code) parts
```

Comment:
Just as a note: This is what I was talking about when I said there was existing code to fix issue #608. Lines 1550-1547 appear to be designed to do exactly what's being done in PR #645, only it occurs as rules are being validated rather than later on when the rules are being applied to the instance definition. The `fixedQuantity` property of this class is never set before its referenced on line 1543 though, which is why values were still being overwritten.

Reply:
Good find! It looks like this bit of code prevents overwriting from happening when we are assigning a value on a Profile or an Extension, which sets the `fixedQuantity` or `patternQuantity` field. The bug you fixed only seemed to occur when we were assigning a value on an Instance. So it looks to me like this little bit of code is working for the purpose it was meant to, Profiles and Extensions, but we overlooked making things work similarly on Instances.